servers/services/src/main/java/org/projectnessie/services/rest/TreeResourceWithAuthorizationChecks.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.services.rest;
+
+import static org.projectnessie.model.Operation.Delete;
+import static org.projectnessie.model.Operation.Put;
+
+import javax.enterprise.context.RequestScoped;
+import javax.enterprise.inject.Default;
+import javax.inject.Inject;
+import org.projectnessie.api.params.CommitLogParams;
+import org.projectnessie.api.params.EntriesParams;
+import org.projectnessie.error.NessieConflictException;
+import org.projectnessie.error.NessieNotFoundException;
+import org.projectnessie.model.Branch;
+import org.projectnessie.model.CommitMeta;
+import org.projectnessie.model.Contents;
+import org.projectnessie.model.Contents.Type;
+import org.projectnessie.model.EntriesResponse;
+import org.projectnessie.model.LogResponse;
+import org.projectnessie.model.Merge;
+import org.projectnessie.model.Operations;
+import org.projectnessie.model.Reference;
+import org.projectnessie.model.Tag;
+import org.projectnessie.model.Transplant;
+import org.projectnessie.services.authz.AccessChecker;
+import org.projectnessie.services.config.ServerConfig;
+import org.projectnessie.versioned.BranchName;
+import org.projectnessie.versioned.NamedRef;
+import org.projectnessie.versioned.TagName;
+import org.projectnessie.versioned.VersionStore;
+
+/** Does authorization checks (if enabled) on the {@link TreeResource} endpoints. */
+@RequestScoped
+@Default
+public class TreeResourceWithAuthorizationChecks extends TreeResource {
```

Comment:
This is missing checks for `TreeApi.getAllReferences` and `getReferenceByName`

Reply:
@laurentgo were those left out from the original design doc on purpose? My assumption was that those were left out on purpose as you still want users to be able to retrieve all available references / a single reference.

Reply:
talked to @laurentgo yesterday and not having checks for `getAllReferences` and `getReferenceByName` was an oversight and didn't make it into the design doc. I updated the design doc and the SPI and all the other related stuff